Subject: daemon: Build with -pthread
From: Richard W.M. Jones rjones@redhat.com Sat Sep 11 09:27:17 2021 +0100
Date: Sat Sep 11 09:27:17 2021 +0100:
Git: e26cfa445a1947503a03c55d8d65263530747062

The daemon failed to link with glibc < 2.34 because we didn't include
the separate pthread library.  Adding -pthread fixes this.

Note this change was also make upstream in commit 733d2182b6 ("Remove
the tools.") although I think either by accident or unrelated.

Reported-by: Toolybird
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2003326

diff --git a/daemon/Makefile.am b/daemon/Makefile.am
index 86aa920e8..b670bb1e2 100644
--- a/daemon/Makefile.am
+++ b/daemon/Makefile.am
@@ -255,6 +255,7 @@ guestfsd_CPPFLAGS = \
 	-I$(top_srcdir)/common/utils \
 	-I$(top_builddir)/common/utils
 guestfsd_CFLAGS = \
+	-pthread \
 	$(WARN_CFLAGS) $(WERROR_CFLAGS) \
 	$(RPC_CFLAGS) \
 	$(AUGEAS_CFLAGS) \
